File: src/main/java/com/example/healthylifestyleapp/HomeScreenActivity.java
package com.example.healthylifestyleapp;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.os.Handler;
import android.widget.ProgressBar;
import android.widget.TextView;

public class HomeScreenActivity extends AppCompatActivity {
    ProgressBar progressBarDrink,progressBarSleep,progressBarFood;
    private int pStatus = 0;
    private Handler handler = new Handler();
    private TextView textViewProgressDrink,textViewProgressFood,textViewProgressSleep;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_home_screen);
        progressBarDrink = (ProgressBar) findViewById(R.id.progressBarDrink);
        progressBarFood=findViewById(R.id.progressBarFood);
        progressBarSleep=findViewById(R.id.progressBarSleep);

        textViewProgressDrink=findViewById(R.id.textViewProgressDrink);
        textViewProgressFood=findViewById(R.id.textViewProgressFood);
        textViewProgressSleep=findViewById(R.id.textViewProgressSleep);

        new Thread(new Runnable() {
            @Override
            public void run() {
                while (pStatus <= 30) {
                    handler.post(new Runnable() {
                        @Override
                        public void run() {
                            progressBarDrink.setProgress(pStatus);
                            textViewProgressDrink.setText(pStatus + " ltr");
                        }
                    });
                    try {
                        Thread.sleep(100);
                    } catch (InterruptedException e) {
                        e.printStackTrace();
                    }
                    pStatus++;
                }
                while (pStatus <= 40) {
                    handler.post(new Runnable() {
                        @Override
                        public void run() {
                            progressBarFood.setProgress(pStatus);
                            textViewProgressFood.setText(pStatus + "" +
                                    "" +
                                    "g");
                        }
                    });
                    try {
                        Thread.sleep(100);
                    } catch (InterruptedException e) {
                        e.printStackTrace();
                    }
                    pStatus++;

                }
                while (pStatus <= 9.40) {
                    handler.post(new Runnable() {
                        @Override
                        public void run() {
                            progressBarSleep.setProgress(pStatus);
                            textViewProgressSleep.setText(pStatus + " ltr");
                        }
                    });
                    try {
                        Thread.sleep(100);
                    } catch (InterruptedException e) {
                        e.printStackTrace();
                    }
                    pStatus++;
                }
            }

        }).start();

    }
    }

File: app/src/main/java/com/example/healthylifestyleapp/ui/activities/HomeScreenActivity.kt
package com.example.healthylifestyleapp.ui.activities

import androidx.appcompat.app.AppCompatActivity

import android.os.Bundle
import android.os.Handler
import android.widget.TextView
import com.example.healthylifestyleapp.R
import kotlinx.android.synthetic.main.activity_home_screen.*

class HomeScreenActivity : AppCompatActivity() {

    private var pStatus = 0
    private val handler = Handler()
    private var textViewProgressDrink: TextView? = null
    private var textViewProgressFood: TextView? = null
    private var textViewProgressSleep: TextView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_home_screen)


        textViewProgressDrink = findViewById(R.id.textViewProgressDrink)
        textViewProgressFood = findViewById(R.id.textViewProgressFood)
        textViewProgressSleep = findViewById(R.id.textViewProgressSleep)

        Thread(Runnable {
            while (pStatus <= 30) {
                handler.post {
                    progressBarDrink.progress = pStatus
                    textViewProgressDrink!!.text = "$pStatus ltr"
                }
                try {
                    Thread.sleep(100)
                } catch (e: InterruptedException) {
                    e.printStackTrace()
                }

                pStatus++
            }
            while (pStatus <= 40) {
                handler.post {
                    progressBarFood.progress = pStatus
                    textViewProgressFood!!.text = pStatus.toString() + "" +
                            "" +
                            "g"
                }
                try {
                    Thread.sleep(100)
                } catch (e: InterruptedException) {
                    e.printStackTrace()
                }

                pStatus++

            }
            while (pStatus <= 9.40) {
                handler.post {
                    progressBarSleep.progress = pStatus
                    textViewProgressSleep!!.text = "$pStatus ltr"
                }
                try {
                    Thread.sleep(100)
                } catch (e: InterruptedException) {
                    e.printStackTrace()
                }

                pStatus++
            }
        }).start()

    }
}

